addons/controllers/csi/vspherecsiconfig_datavalues.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2022 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package controllers
+
+// DataValues is the data values type generated by the VSphereCSIConfig CR
+type DataValues struct {
+	VSphereCSI   *DataValuesVSphereCSI   `yaml:"vsphereCSI,omitempty"`
+	VSpherePVCSI *DataValuesVSpherePVCSI `yaml:"vspherePVCSI,omitempty"`
+}
+
+type DataValuesVSpherePVCSI struct {
+	ClusterName                      string `yaml:"clusterName"`
+	ClusterUID                       string `yaml:"clusterUID"`
+	Namespace                        string `yaml:"namespace"`
+	SupervisorMasterEndpointHostname string `yaml:"supervisorMasterEndpointHostname"`
+	SupervisorMasterPort             int32  `yaml:"supervisorMasterPort"`
+}
+
+type DataValuesVSphereCSI struct {
```

Comment:
question: do we want `omitempty` in this struct ?

Reply:
yes. values.yaml should only have 'vsphereCSI' or 'vspherePVCSI'. Whichever is defined.